runtime/src/main/java/org/corfudb/runtime/collections/TxnContextShim.java

```diff
@@ -230,6 +608,92 @@ void putRecord(@Nonnull Table<K, V, M> table,
+    }
+
+    /**
+     *
+     * Delete a record given just tableName and the metadata for validation against.
+     *
+     * @param tableName - full string representation of the table
+     * @param key - the key of the record being inserted
+     * @param metadata - the metadata which will be validated
+     * @param <K> - type of the key or identifier.
+     * @param <V> - type of the value or payload
+     * @param <M> - type of the metadata
+     */
+    public <K extends Message, V extends Message, M extends Message>
+    void deleteRecord(@Nonnull String tableName,
+                      @Nonnull final K key,
+                      @Nullable final M metadata) {
+        deleteRecord(this.getTable(tableName), key, metadata);
+    }
+
+    static class MergeCallbackForDeleteImpl implements TxnContext.MergeCallback {
+        /**
+         * Core logic for handling metadata modifications on delete with metadata.
+         *
+         * @param table       - The table object on which the merge is being done.
+         * @param key         - Key of the record on which merge is being done.
+         * @param oldRecord   - The previous fetched record that exists in the table.
+         * @param deltaUpdate - Record with only the metadata passed in for validation.
+         * @param <K>         - type of the key
+         * @param <V>         - type of the value or payload
+         * @param <M>         - type of the metadata
+         * @return null if validation was successful or StaleRevision exception if not.
+         */
+        @Override
+        public <K extends Message, V extends Message, M extends Message>
+        CorfuRecord<V, M> doMerge(Table<K, V, M> table,
+                                  K key,
+                                  CorfuRecord<V, M> oldRecord,
+                                  CorfuRecord<V, M> deltaUpdate) {
+            M deltaMetadata = deltaUpdate.getMetadata();
+            if (deltaUpdate.getPayload() != null) {
+                throw new IllegalArgumentException("Non-null payload sent for delete validation");
+            }
+
+            final Message.Builder builder = deltaMetadata.toBuilder();
+            for (Descriptors.FieldDescriptor fieldDescriptor : deltaMetadata.getDescriptorForType().getFields()) {
+                if ("revision".equals(fieldDescriptor.getName())) {
+                    if (oldRecord == null || oldRecord.getMetadata() == null) {
+                        return null;
+                    } else {
+                        Long prevRevision = (Long) oldRecord.getMetadata().getField(fieldDescriptor);
+                        Long givenRevision = (Long) deltaMetadata.getField(fieldDescriptor);
+                        if ((givenRevision > 0 && // Validate revision only if set
+                                prevRevision.longValue() == givenRevision.longValue())
+                                || givenRevision == 0) { // Do not validate revision if field isn't set
```

Comment:
nit:since you use short circuit operator || the faster check could be first.

Reply:
done. thanks